set_required_fields.php

```diff
@@ -25,26 +25,44 @@
 
 
 use profilefield_branching\form\set_require_fields_form;
+use profilefield_branching\utility;
 
 require_once(__DIR__.'/../../../../config.php');
 
+// Not the full require_login() check, as it creates a cyclic loop around profilefield_branching_after_require_login().
+if ((!isloggedin() or isguestuser())) {
```

Comment:
I think this is problematic, and probably won't work with saml, or might add extra redirects. We should simply not redirect if we are already on this page which is how other plugins like tool_securityquestions work

Reply:
I've removed the redirects, and it now prompts a page with a language string, then dies.